docs/Style-guide.md

```diff
@@ -15,6 +15,25 @@ Principles
 
 General guide lines
 -------------------
+### Code linting and formatting tools
+`stylish-haskell` is used for automatic code formatting and `hlint` for linting.
+They are configured using `.stylish-haskell.yml` and `.hlint.yml` files in the repository root.
+If you would like to check your changes before committing them, you can use the following `.git/hooks/pre-commit` hook:
+```bash
+#!/bin/sh
+
+for x in $(git diff --staged --name-only --diff-filter=ACM | tr '\n' ' '); do
+  if [ "${x##*.}" = "hs" ]; then
+    stylish-haskell -i "$x"
+    hlint "$x"
+  fi
+done
+
+# fail if there are style issues
+git --no-pager diff --exit-code
+```
```

Comment:
Can we diff against the most recent common commit with `origin/master` instead?  Ideally it should capture all files 
changed in the branch and not just staged files.

Also wondering if it can actually be installed as a git hook and not be in the wiki.

Reply:
Yeah, I think it would be good to at least have it as a script in the `scripts` directory and then we can reference that script from here instead.

Reply:
I've created PR then: https://github.com/input-output-hk/cardano-node/pull/5166